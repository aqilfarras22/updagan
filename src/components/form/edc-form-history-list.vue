<template>
  <table-component
    showEmpty
    noGutters
    :items="history"
    :fields="formHistoryFields"
  >
    <!-- date -->
    <template slot="createdAt" slot-scope="data">
      {{ new Date(data.row.item.createdAt) | formatUTCDate }}
    </template>
    <!-- action -->
    <template slot="eventType" slot-scope="data">
      {{ data.row.item.primaryReason && data.row.item.primaryReason!='' ? data.row.item.primaryReason : (verificationStatusText[data.row.item.eventType] || approvalStatusText[data.row.item.eventType] || historyStatusText[data.row.item.eventType]) }}
    </template>
    <!-- view -->
    <template slot="view" slot-scope="data">
      <a v-if="historyStatus.MODIFIED === data.row.item.eventType" @click.prevent="view(data.row.item)" href="">View</a>
    </template>
  </table-component>
</template>

<script>
import TableComponent from '@/components/common/table';
import {
  approvalStatusText,
  verificationStatusText,
  historyStatusText,
  historyStatus,
} from '@/constants/form';

export default {
  name: 'form-history-list',
  components: {
    TableComponent,
  },
  props: {
    history: {
      type: Array,
      default: null,
    },
  },
  data() {
    return {
      approvalStatusText,
      verificationStatusText,
      historyStatusText,
      historyStatus,
      formHistoryFields: [
        {
          key: 'createdAt',
          label: 'Date',
          sortable: true,
        },
        {
          key: 'eventType',
          label: 'Action / Changes',
          sortable: true,
        },
        {
          key: 'createdBy',
          label: 'By',
          sortable: true,
        },
        {
          key: 'view',
          label: ' ',
        },
      ],
    };
  },
  methods: {
    view(item) {
      this.$emit('view', item);
    },
  },
};
</script>
