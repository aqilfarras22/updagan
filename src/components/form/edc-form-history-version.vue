<template>
  <div>
    <b-row class="border-bottom header-row">
      <b-col><a @click.prevent="back" class="backButton" href=""><i class="fa fa-caret-left fa-lg"></i>&nbsp; Back to history</a></b-col>
      <b-col class="text-right"><i class="square"></i>&nbsp; <span class="info">Edited fields</span></b-col>
    </b-row>
    <div>
      <b-card-group v-if="!edcFormVersionData.isUnSchedule" class="pb-15">
        <b-card no-body class="border-0" :header="edcFormVersionData.name">
          <div v-if="edcFormVersionData.primaryReason" class="reason">
            Reason for change: {{ edcFormVersionData.primaryReason }}
          </div>
          <SubmittedFormPreview v-if="edcFormVersionData.fields" :isHistory="true" :formData="edcFormVersionData.fields" />
        </b-card>
      </b-card-group>
    </div>
    <div v-if="edcFormVersionData.isUnSchedule && edcFormVersionData.fields">
      <b-card-group v-for="(unscheduleAnswers, index) in edcFormVersionData.fields" :key="index" :class="{'border-top': index!=0 ,'pb-15' : index+1 === edcFormVersionData.fields.length, 'pb-5' : index+1 !== edcFormVersionData.fields.length}">
        <b-card no-body class="border-0" :header="unscheduleAnswers.formName">
          <div v-if="edcFormVersionData.primaryReason" class="reason">
            Reason for change: {{ edcFormVersionData.primaryReason }}
          </div>
          <SubmittedFormPreview v-if="unscheduleAnswers.formData" :isHistory="true" :formData="unscheduleAnswers.formData"/>
        </b-card>
      </b-card-group>
    </div>
  </div>
</template>

<script>
import SubmittedFormPreview from '@/components/edc-form/submitted-form-preview';

export default {
  name: 'form-history-version',
  components: { SubmittedFormPreview },
  props: {
    edcFormVersionData: Object,
  },
  methods: {
    back() {
      this.$emit('back');
    },
  },
};
</script>

<style scoped>
.header-row {
  padding: 0.9rem 0 0.65rem;
}
.square {
  display: inline-block;
  width: 16px;
  height: 16px;
  border-radius: 1px;
  margin-bottom: -2px;
  background-color: #ffa200;
}
.info {
  font-size: 12px;
  display: inline-block;
}
.reason {
  font-family: "RobotoBold" !important;
  color: #666669;
  padding-left: 30px;
  padding-top: 19px;
}
</style>
