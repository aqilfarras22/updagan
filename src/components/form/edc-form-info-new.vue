<template>
  <div class="edc-form-bar">
	<div class="row">
		<div class="col-md-2">
			<div class="heading">Form/Packet Name</div>
			<div class="value">{{ clearFormName() }}</div>
		</div>
		<div class="col-md-2">
			<div class="heading">Verification Status</div>
			<div class="value" :class="edcFormInfo.verificationStatus">{{ verificationStatusText[edcFormInfo.verificationStatus] }}</div>
			<div v-if="verified && edcFormInfo.verifiedOn" class="date value">
				({{ new Date(edcFormInfo.verifiedOn) | formatGMTDate }})
			</div>
		</div>
		<div class="col-md-2">
			<div class="heading">Approval Status</div>
			<div class="value" :class="edcFormInfo.approvalStatus">{{ approvalStatusText[edcFormInfo.approvalStatus] }}</div>
			<span v-if="approved && edcFormInfo.approvedOn" class="date value">
			({{ new Date(edcFormInfo.approvedOn) | formatGMTDate }})
			</span>
		</div>
	</div>
  </div>
</template>

<script>
import { AclRule } from 'vue-acl';
import { mapGetters } from 'vuex';
import {
  approvalStatusText,
  verificationStatusText,
  verificationStatus,
  approvalStatus,
} from '@/constants/form';

export default {
  name: 'form-info',
  components: {},
  props: {
    edcFormInfo: {
		type: Object,
		default: {}
	}
  },
  data() {
    return {
      approvalStatusText,
	  verificationStatusText,
	  verificationStatus
	};
  },
  computed: {
    ...mapGetters(['user']),
    userAccess() {
      return new AclRule(this.user.userRole).generate();
    }
  },
  methods: {
	clearFormName() {
    	return this.edcFormInfo.formName ? this.edcFormInfo.formName.replace(/_/g, ' ') : '';
	},
	verified() {
      	return this.edcFormInfo.verificationStatus === verificationStatus.VERIFIED;
    },
    approved() {
      	return this.edcFormInfo.approvalStatus === approvalStatus.APPROVED;
    }
  }
};
</script>

<style scoped lang="scss">
.edc-form-bar {
	height: 60px;
  background-color: #fff;
  padding: 10px 0px;
  -webkit-box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.06);
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.06);
  margin-top: 15px;
  height: 75px;
  .heading {
    font-weight: 600;
  }
}
.date {
  font-size: 12px;
  line-height: 1.67;
}
.telehealth-call {
  .heading {
    font-size: 10px;
  }
  .value {
    font-size: 10px;
  }
}
</style>
