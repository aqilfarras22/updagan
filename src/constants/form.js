export const fieldTypes = {
  POLAR_QUESTION: 'POLAR_QUESTION',
  DROPDOWN: 'DROPDOWN',
  DATEPICKER: 'DATEPICKER',
  INPUT: 'INPUT',
  LABEL: 'LABEL',
};

export const formFieldQueryStatus = {
  OPEN: 'OPEN',
  CLOSE: 'CLOSE',
};

export const historyStatus = {
  MODIFIED: 'MODIFIED',
};

export const historyStatusText = {
  MODIFIED: 'Original Submission',
};

// approve
export const approvalStatus = {
  APPROVED: 'APPROVED',
  NOT_APPROVED: 'NOT_APPROVED',
};
export const approvalStatusText = {
  APPROVED: 'Approved',
  NOT_APPROVED: 'Unapproved',
};

// verification
export const verificationStatus = {
  VERIFIED: 'VERIFIED',
  NOT_VERIFIED: 'NOT_VERIFIED',
};
export const verificationStatusText = {
  VERIFIED: 'Verified',
  NOT_VERIFIED: 'Unverified',
};
