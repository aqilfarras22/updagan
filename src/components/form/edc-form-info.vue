<template>
  <div v-if="formId">
    <dl v-if="formName" ref="formName">
      <dt>Form/Packet Name</dt>
      <dd>{{ clearFormName }}</dd>
    </dl>
    <dl ref="verificationStatus">
      <dt>Verification Status</dt>
      <dd>
        <div :class="verificationStatus">{{ verificationStatusText[verificationStatus] }}</div>
        <div v-if="verified && verifiedOn" class="date">
          ({{ new Date(verifiedOn) | formatGMTDate }})
        </div>
      </dd>
    </dl>
    <dl ref="approvalStatus">
      <dt>Approval Status</dt>
      <dd>
        <div :class="approvalStatus">{{ approvalStatusText[approvalStatus] }}</div>
        <span v-if="approved && approvedOn" class="date">
          ({{ new Date(approvedOn) | formatGMTDate }})
        </span>
      </dd>
    </dl>
    <div class="actions">
      <button-cmp v-if="canPiAccess || canSubPiAccess || canScAccess || canHomeHealthAccess" class="form-action-button" size="lg" variant="primary" @click="openApprovalModal">{{ approvalBtnText }}</button-cmp>
      <button-cmp v-if="canCraAccess" class="form-action-button" size="lg" variant="primary" @click="openVerificationModal">{{ verifyBtnText }}</button-cmp>
    </div>
    <formApprovalModal
      v-if="canPiAccess || canSubPiAccess || canScAccess || canHomeHealthAccess"
      ref="approvalModal"
      callApi
      :formId="formId"
      :approved="approved"
      @confirmed="approvalConfirmed"
    />
    <formVerificationModal
      v-if="canCraAccess"
      ref="verificationModal"
      callApi
      :formId="formId"
      :verified="verified"
      @confirmed="verificationConfirmed"
    />
  </div>
</template>

<script>
import { AclRule } from 'vue-acl';
import { mapGetters } from 'vuex';
import formVerificationModal from '@/components/modals/edc-form-verification-modal';
import formApprovalModal from '@/components/modals/edc-form-approval-modal';
import ButtonCmp from '@/components/common/button';

import { PI, CRA, SUB_PI, STUDY_COORDINATOR, HOME_HEALTH } from '@/constants/authRoles';
import {
  approvalStatusText,
  verificationStatusText,
  verificationStatus,
  approvalStatus,
} from '@/constants/form';

export default {
  name: 'form-info',
  components: {
    formVerificationModal,
    formApprovalModal,
    ButtonCmp,
  },
  props: {
    formId: String,
    formName: String,
    approvalStatus: String,
    verificationStatus: String,
    verifiedOn: [String, Number],
    approvedOn: [String, Number],
  },
  data() {
    return {
      approvalStatusText,
      verificationStatusText,
    };
  },
  computed: {
    ...mapGetters(['user']),
    userAccess() {
      return new AclRule(this.user.userRole).generate();
    },
    canPiAccess() {
      this.$acl.change(`${PI}`);
      return this.$acl.check('userAccess');
    },
    canSubPiAccess() {
      this.$acl.change(`${SUB_PI}`);
      return this.$acl.check('userAccess');
    },
    canScAccess() {
      this.$acl.change(`${STUDY_COORDINATOR}`);
      return this.$acl.check('userAccess');
    },
    canHomeHealthAccess() {
      this.$acl.change(`${HOME_HEALTH}`);
      return this.$acl.check('userAccess');
    },
    canCraAccess() {
      this.$acl.change(`${CRA}`);
      return this.$acl.check('userAccess');
    },
    clearFormName() {
      return this.formName ? this.formName.replace(/_/g, ' ') : '';
    },
    verified() {
      return this.verificationStatus === verificationStatus.VERIFIED;
    },
    approved() {
      return this.approvalStatus === approvalStatus.APPROVED;
    },
    verifyBtnText() {
      return this.verified ? 'Unverify' : 'Verify';
    },
    approvalBtnText() {
      return this.approved ? 'Unapprove' : 'Approve';
    },
  },
  methods: {
    openVerificationModal() {
      this.$refs.verificationModal.show();
    },
    openApprovalModal() {
      this.$refs.approvalModal.show();
    },
    verificationConfirmed({ verified }) {
      const status = verified ? verificationStatus.VERIFIED : verificationStatus.NOT_VERIFIED;
      this.$emit('verificationConfirmed', status);
    },
    approvalConfirmed({ approved }) {
      const status = approved ? approvalStatus.APPROVED : approvalStatus.NOT_APPROVED;
      this.$emit('approvalConfirmed', status);
    },
  },
};
</script>

<style scoped>
.APPROVED,
.VERIFIED {
  color: #48bf7a !important;
}
.NOT_VERIFIED,
.NOT_APPROVED {
  color: #f16559 !important;
}
.form-action-button {
  width: 100%;
  margin-top: 1px;
  margin-bottom: 1px;
}
dt {
  font-size: 10px;
  line-height: 1.6;
  text-transform: uppercase;
  color: #adaeb1;
}
dd {
  color: #666669;
  font-size: 14px;
  font-weight: 500;
}
.date {
  font-size: 12px;
  line-height: 1.67;
}
</style>
