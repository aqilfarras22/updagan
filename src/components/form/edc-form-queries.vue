<template>
  <b-tabs class="" v-model="tabIndex">
    <b-tab :title="`Open Queries: ${openQueries.length}`" :title-link-class="queriesOpen()">
      <FormQueriesCommentForm v-if="value" @cancel="cancelQuery" @submit="createQuery"/>
      <div v-else>
        <div v-if="openQueryId">
          <FormQueriesQueryDetails
            :userNames="userNames"
            :comments="comments"
            :query="openQuery"
            @back="hideOpenQuery"
            @onQueryCommentCreate="onQueryCommentCreate"
            @onQueryClosed="onQueryClosed"
          />
        </div>
        <div v-else>
          <FormQueriesHeader v-if="filter" :title="`Filtered by field ID ${shortFilterId}`" @back="resetFilter" backTitle="Clear"/>
          <b-row v-if="!openQueries.length" class="py-2">
            <b-col><p>No queries created yet.<br>Select a field to create a query.</p></b-col>
          </b-row>
          <FormQueriesList
            v-else
            :queries="openQueries"
            @view="viewOpenQuery"
          />
        </div>
      </div>
    </b-tab>
    <b-tab :title="`Closed Queries: ${closedQueries.length}`">
      <div v-if="closedQuery">
        <FormQueriesQueryDetails
          closed
          :comments="comments"
          :userNames="userNames"
          :query="closedQuery"
          @back="hideClosedQuery"
          @onQueryReopened="onQueryReopened"
        />
      </div>
      <div v-else>
        <FormQueriesHeader v-if="filter" :title="`Filtered by field ID ${filter}`" @back="resetFilter" backTitle="Clear"/>
        <b-row v-if="!closedQueries.length">
          <b-col class="py-2"><p>No queries closed yet.</p></b-col>
        </b-row>
        <FormQueriesList
          v-else
          :queries="closedQueries"
          @view="viewClosedQuery"
        />
      </div>
    </b-tab>
  </b-tabs>
</template>

<script>
import sortBy from 'lodash/sortBy';
import FormQueriesList from '@/components/form/edc-form-queries/form-queries-list';
import FormQueriesQueryDetails from '@/components/form/edc-form-queries/form-queries-query-details';
import FormQueriesCommentForm from '@/components/form/edc-form-queries/form-queries-comment-form';
import FormQueriesHeader from '@/components/form/form-queries-header';

import { formFieldQueryStatus } from '@/constants/form';

export default {
  name: 'edc-form-queries',
  components: {
    FormQueriesList,
    FormQueriesQueryDetails,
    FormQueriesCommentForm,
    FormQueriesHeader,
  },
  props: {
    value: String,
    queries: Array,
    filter: String,
    formId: String,
    studyId: String,
    comments: Array,
  },
  data() {
    return {
      tabIndex: 0,
      openQueryId: null,
      closedQueryId: null,
      userNames: {},
    };
  },
  async mounted() {
    const type="studyteam"
    const associated = "true"
    const studyTeamData = await this.$api.getStudyUsersFact(this.$route.params.studyId, type, associated);
    const siteTeamData = await this.$api.getUserNames(this.$route.params.studyId);
    studyTeamData.body.forEach(team => {
      this.userNames[team.id] = `${team.firstName} ${team.lastName}`;
    });
    siteTeamData.body.forEach(team => {
      this.userNames[team.id] = `${team.firstName} ${team.lastName}`;
    });
  },
  computed: {
    shortFilterId() {
      if (this.filter) {
        // return this.filter.slice(0, 5).toUpperCase();
        return this.filter;
      }
      return '';
    },
    filteredQueries() {
      let filtered = this.queries;
      if (this.filter) {
        filtered = this.queries.filter(q => q.fieldId === this.filter);
        this.$forceUpdate();
      }
      return sortBy(filtered, ['openedOn']).reverse();
    },
    openQueries() {
      return this.filteredQueries.filter(q => q.status === formFieldQueryStatus.OPEN);
    },
    closedQueries() {
      return this.filteredQueries.filter(q => q.status !== formFieldQueryStatus.OPEN);
    },
    openQuery() {
      return this.queries.find(q => q.queryId === this.openQueryId);
    },
    closedQuery() {
      return this.queries.find(q => q.queryId === this.closedQueryId);
    },
  },
  watch: {
    value(val) {
      if (val) {
        this.openQueryId = null;
        this.closedQueryId = null;
        this.tabIndex = 0;
      }
    },
    filter(val) {
      if (!this.value) {
        if (val) {
          this.shortFilterId = val;
          this.openQueryId = null;
          this.closedQueryId = null;
          this.tabIndex = 0;
        }
      }
    },
  },
  methods: {
    queriesOpen() {
      return this.filteredQueries.filter(q => q.status === formFieldQueryStatus.OPEN).length > 0 ? 'text-warning' : '';
    },
    viewOpenQuery(queryId) {
      this.openQueryId = queryId;
      const field = this.openQueries.find(q => q.queryId === queryId);
      this.$emit('updateComment', queryId, field.fieldId);
    },
    hideOpenQuery() {
      this.$emit('updateQueries');
      this.openQueryId = null;
      this.$emit('resetFilter');
    },
    viewClosedQuery(queryId) {
      this.closedQueryId = queryId;
      const field = this.closedQueries.find(q => q.queryId === queryId);
      this.$emit('updateComment', queryId, field.fieldId);
    },
    hideClosedQuery() {
      this.closedQueryId = null;
      this.$emit('resetFilter');
    },
    createQuery(commentData) {
      const newQuery = {
        fieldId: this.value,
        formId: this.formId,
        comment: commentData,
      };
      this.$emit('createQuery', newQuery);
      this.$emit('input', null);
    },
    onQueryCommentCreate(commentData) {
      this.$emit('createComment', commentData);
    },
    cancelQuery() {
      this.$emit('input', null);
      this.$emit('resetFilter');
    },
    resetFilter() {
      this.$emit('resetFilter');
      window.scrollTo(0, 0);
    },
    onQueryClosed(query) {
      this.openQueryId = null;
      this.$emit('onQueryClosed', query);
    },
    onQueryReopened(query) {
      this.closedQueryId = null;
      this.$emit('onQueryReopened', query);
    },
  },
};
</script>
<style>
a.text-warning:hover, a.text-warning:focus, .text-warning {
  color: #f5a623 !important;
}
</style>
