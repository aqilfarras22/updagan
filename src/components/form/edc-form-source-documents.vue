<template>
<div>
  <div v-if="!canSiteDataLockAccess" class="add-document"><div class="plus-sign pull-left" @click="openUploadModal()"><span class="plus">+</span></div><div class="pull-left cursor_pointer"  @click="openUploadModal()">Add a source document</div><div class="pull-left max-size">Max file size: 100mb</div>
  <div style="clear:both;"></div>
  </div>
  <b-modal ref="upload-modal" hide-footer hide-header>
                <div class="d-block text-center upload-modal">
                  <div class="description">If you are uploading Protected Health Information (PHI), please make sure all Personally Identifiable Information (PII) is redacted or removed.</div>
                  <div class="btn-grp mt-30">
                     <div class="file-upload understand-button">
  <input type="file" id="myfile" name="myfile" class="file-input" @change="selectedFile()" @click="closeUploadModal()"/>I understand</div>
                  </div>
    </div>
  </b-modal>
  <b-modal ref="progress-modal" hide-footer hide-header>
                <div class="d-block text-center progress-modal">
                  <div class="upload-file">Uploading file...</div>
                  <img src="static/img/icn-csv.png"/>
                  <div class="pb-15">{{fileName}}</div>
                  <progress id="progressBar" value="0" max="100" class="progress"></progress>
                  <div class="uploading-now">Uploading now. Thanks for being patient.</div>
                  <div class="uploading-now delete pt-40"><span class="cursor_pointer" @click="cancelUpload()">Cancel upload</span></div>

    </div>
  </b-modal>
  <table-component
    showEmpty
    noGutters
    :items="documents"
    :fields="formHistoryFields"
  >


     <template slot="name" slot-scope="data">
      {{ data.row.item.name }}
    </template>
     <template slot="createdAt" slot-scope="data">
      {{ new Date(data.row.item.createdAt) | formatUTCDate }}
    </template>
     <template slot="uploadedBy" slot-scope="data">
      {{ data.row.item.createdByName }} ({{data.row.item.createdByUserRole}})
    </template>
    <template slot="size" slot-scope="data">
      {{ data.row.item.fileSize }}
    </template>
     <template slot="Download" slot-scope="data">
      <a @click="downloadDocument(data.row.item.id)" class="download cursor_pointer">Download</a>
    </template>
    <template slot="Delete" slot-scope="data">
      <a @click="deleteDocument(data.row.item.id)" class="delete cursor_pointer">Delete</a>
    </template>

  </table-component>
  </div>
</template>

<script>
import TableComponent from '@/components/common/table';
import swal from 'sweetalert2';

import {
  approvalStatusText,
  verificationStatusText,
  historyStatusText,
  historyStatus,
} from '@/constants/form';

import { SITEDATALOCK, PI } from "@/constants/authRoles";
import { mapGetters } from "vuex";
import { AclRule } from "vue-acl";
import toastr from 'toastr';

export default {
  name: 'edc-form-source-documents',
  components: {
    TableComponent,
  },
  props: [
    'documents',
    'studyId',
    'participantId',
    'responseId',
    'type',
    'participantInfo'
  ],
  data() {
    return {
      xmlHTTP: '',
      uploadUrl: '',
      s3Id: '',
      fileName: '',
      approvalStatusText,
      verificationStatusText,
      historyStatusText,
      historyStatus,
      formHistoryFields: [
        {
          key: 'name',
          label: 'NAME',
          sortable: true,
        },
        {
          key: 'createdAt',
          label: 'UPLOADED',
          sortable: true,
        },
        {
          key: 'uploadedBy',
          label: 'BY',
          sortable: false,
        },
        {
          key: 'fileSize',
          label: 'FILE SIZE',
          sortable: false,
        },
        {
          key: 'Download',
          label: 'ACTIONS'
        },
        {
          key: 'Delete',
          label: ' '
        }
      ],
    };
  },
  mounted() {
  },
  computed: {
    ...mapGetters(["user"]),
    userAccess() {
      return new AclRule(this.user.userRole).generate();
    },
    canPiAccess() {
      this.$acl.change(`${PI}`);
      return this.$acl.check("userAccess");
    },
    canSiteDataLockAccess() {
      this.$acl.change(`${SITEDATALOCK}`);
      return this.$acl.check("userAccess");
    }
  },
  methods: {
    getSourceDocuments() {
      this.$emit('getDocuments')
  },
    openUploadModal() {
      this.$refs['upload-modal'].show()
  },
  selectedFile() {
     this.$api.getPreSignedUrlFact(this.studyId, this.participantId, this.responseId, this.type, this.participantInfo.siteId)
        .then(response => {
          this.uploadUrl= response.body.preSignedUrl;
          this.s3Id= response.body.s3Id;
          document.getElementById("progressBar").value = 0
          var selectedFile = document.getElementById("myfile");
                var file = selectedFile.files[0];
                if (file) {
                  if(file.size < 104857600) {
                  this.$refs['progress-modal'].show()
                  var url = this.uploadUrl;
                  this.fileName = file.name;
                  this.$forceUpdate()
                  this.xmlHTTP= new XMLHttpRequest();
                  // xmlHTTP.upload.addEventListener("loadstart", loadStartFunction, false);
                  this.xmlHTTP.upload.addEventListener("progress", this.progressFunction, false);
                  this.xmlHTTP.addEventListener("load", this.transferCompleteFunction, false);
                  this.xmlHTTP.addEventListener("error", this.uploadFailed, false);
                  this.xmlHTTP.addEventListener("abort", this.uploadCanceled, false);
                  this.xmlHTTP.open("PUT", url, true);
                  this.xmlHTTP.setRequestHeader("Content-Type", file.type);
                  //xmlHTTP.setRequestHeader('book_id','10');
                  this.xmlHTTP.send(file);
                  }
                  else{
                    this.$toastr.error('File size exceeds the maximum of 100 MB.')
                    document.getElementById("myfile").value = null;
                  }
                }
        },
        error => {
          this.$toastr.error(
            error.body && error.body.message
              ? error.body.message
              : 'Error while uploading the document.',
          );
        })

              },
            progressFunction(evt) {
                var progressBar = document.getElementById("progressBar");
                var percentageDiv = document.getElementById("percentageCalc");
                if (evt.lengthComputable) {
                    progressBar.max = evt.total;
                    progressBar.value = evt.loaded;
                    // percentageDiv.innerHTML = Math.round(evt.loaded / evt.total * 100) + "%";
                }
            },
            transferCompleteFunction(evt) {
              let fileDetails = {}
              fileDetails.name = this.fileName
              fileDetails.s3Id = this.s3Id
              fileDetails.description = ''
              fileDetails.type = this.type
              this.$api.postUploadFact(this.studyId, this.participantId, this.responseId, fileDetails, this.participantInfo.siteId)
        .then(response => {
          this.getSourceDocuments()
        },
        error => {
          this.$toastr.error(
            error.body && error.body.message
              ? error.body.message
              : 'Error while uploading the document.',
          );
        })
        this.$refs['progress-modal'].hide()
        document.getElementById("myfile").value = null;
            },
        uploadFailed() {
          this.$toastr.error('Error while uploading the document.')
        },
        downloadDocument(id) {
          if (!this.canSiteDataLockAccess) {
              this.$api.getDownloadUrlFact(this.studyId, this.participantId, this.responseId, id, this.participantInfo.siteId)
          .then(response => {
            this.downloadUrl = response.body.preSignedUrl
            window.open(this.downloadUrl, "_blank")
          },
          error => {
            this.$toastr.error(
              error.body && error.body.message
                ? error.body.message
                : 'Error while downloading the document.',
            );
          })
          } else {
            toastr.warning('Access restricted');
          }        
        },
        deleteDocument(id) {
          if (!this.canSiteDataLockAccess) {
                swal({
            title:
              'Are you sure you want to delete the document?',
            type: 'warning',
            showCancelButton: true,
            confirmButtonColor: '#b8c7ce',
            cancelButtonColor: '#34537d',
            confirmButtonText: 'Cancel',
            cancelButtonText: 'Yes',
          }).then(({ dismiss }) => {
            if (dismiss === 'cancel') {
            this.$api.deleteDocumentFact(this.studyId, this.participantId, this.responseId, id, this.participantInfo.siteId)
            .then(response => {
              this.getSourceDocuments()
            },
            error => {
              this.$toastr.error(
                error.body && error.body.message
                  ? error.body.message
                  : 'Error while deleting the document.',
              );
            })
            }
          }); 
         } else {
            toastr.warning('Access restricted');
        }        
        },
        cancelUpload(){
          this.xmlHTTP.abort()
          this.$refs['progress-modal'].hide()
          document.getElementById("myfile").value = null;
        },
        closeUploadModal() {
        this.$refs['upload-modal'].hide()
        }
  }
};
</script>
<style lang="scss" scoped>
.add-document {
  padding: 20px;
  color: #4c8ce4;
  font-weight: bold;
}

.progress {
  width: 400px;
  margin: 0 auto;
  height: 5px!important;
}

.download {
  color: #4c8ce4!important;
}

.delete {
  color: #f16559!important;
}

.upload-modal {
  padding: 40px;
  font-weight: bold;
}

.understand-button {
  background-color: #4c8ce4!important;
  padding: 10px 80px;
  color: #ffffff!important;
  border-radius: 100px;
  font-weight: normal!important;
}

.plus-sign {
  width: 17px;
  height: 17px;
  border-radius: 100px;
  background-color: #4c8ce4;
  color: #ffffff;
  text-align: center;
  margin-right: 5px;
  position: relative;
  top: 2px;
  cursor: pointer;
}

.plus {
  position: relative;
  bottom: 2.5px;
}

.progress-modal {
  padding: 20px;
}

.upload-file {
  font-size: 17px;
  font-weight: bold;
  margin-bottom: 30px;
}

.uploading-now {
  font-size:12px;
  padding-top: 10px;
  color: #666669;
}

progress::-webkit-progress-value {
  background: #4c8ce4!important;
}

.max-size {
  color: grey;
  font-size: 12px;
  font-weight: normal;
  margin: 3px 0 0 10px;
}

</style>
