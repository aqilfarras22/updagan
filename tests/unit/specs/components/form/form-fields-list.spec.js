import { mount, shallowMount, createLocalVue } from '@vue/test-utils';
import { fieldTypes } from '@/constants/form';
import BootstrapVue from 'bootstrap-vue';

import formInfoComponent from '@/components/form/form-fields-list';
import formatDate from '@/constants/filter';

const { it, describe, expect } = global;

const simpleQuestion = {
  fieldId: '28n238n23823',
  question: 'question',
  response: ['answer'],
};

const dateQuestion = {
  fieldId: '28n212121',
  fragmentType: fieldTypes.DATEPICKER,
  question: 'question',
  response: [1519138199],
};

const sections = [
  {
    name: 'section',
    fields: [simpleQuestion, dateQuestion],
  },
];

const itemSelector = '.item';
const openQueryButtonSelector = '.open-query';

const localVue = createLocalVue();
localVue.use(BootstrapVue);
localVue.use({
  install(vue) {
    vue.filter('moment', (v, format) => format);
    vue.filter(formatDate);
  },
});

const createMountedWrapper = (options = {}) => mount(formInfoComponent, {
  localVue,
  propsData: { sections },
  ...options,
});

describe('components/form/form-fields-list.vue', () => {
  it('should create instance', () => {
    const wrapper = createMountedWrapper();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it('should show correct number of questions', () => {
    const wrapper = createMountedWrapper();
    expect(wrapper.findAll(itemSelector).length).toBe(2);
  });

  it('should fire "openQuery" on click corresponding button over item', async () => {
    const wrapper = createMountedWrapper();
    wrapper.find(openQueryButtonSelector).trigger('click');
    await wrapper.vm.$nextTick();
    expect(wrapper.emitted('openQuery')[0][0]).toBe(simpleQuestion.fieldId);
  });

  it('should not fire "select" on clicking item without queries', async () => {
    const wrapper = createMountedWrapper();
    wrapper.find(itemSelector).trigger('click');
    await wrapper.vm.$nextTick();
    expect(wrapper.emitted('selected')).toBe(undefined);
  });

  it('should fire "select" on clicking item with queries', async () => {
    const wrapper = createMountedWrapper({
      propsData: {
        sections,
        fieldsWithQueries: [simpleQuestion.fieldId],
      },
    });
    wrapper.find(itemSelector).trigger('click');

    await wrapper.vm.$nextTick();
    expect(wrapper.emitted('selected')[0][0]).toBe(simpleQuestion.fieldId);
  });

  it('has the expected html structure', () => {
    const wrapper = shallowMount(formInfoComponent, { localVue, propsData: { sections } });
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
