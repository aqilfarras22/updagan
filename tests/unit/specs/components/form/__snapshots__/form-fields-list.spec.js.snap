// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/form/form-fields-list.vue has the expected html structure 1`] = `
<div>
  <b-card-group-stub
    class="border-top"
    tag="div"
  >
    <b-card-stub
      bodytag="div"
      class="border-0"
      footertag="div"
      header="section"
      headertag="div"
      nobody="true"
      subtitletag="h6"
      tag="div"
      titletag="h4"
    >
      <b-list-group-stub
        flush="true"
        tag="div"
      >
        <b-list-group-item-stub
          activeclass="active"
          class="item border-0 [object Object]"
          event="click"
          exactactiveclass="active"
          routertag="a"
          tag="div"
          target="_self"
        >
          <!---->
           
          <dl>
            <dt
              class=""
            >
              
              1. question
            
            </dt>
             
            <dd>
              answer
            </dd>
          </dl>
           
          <button-cmp-stub
            class="open-query"
            variant="outline-primary"
          >
            Open a query
          </button-cmp-stub>
        </b-list-group-item-stub>
        <b-list-group-item-stub
          activeclass="active"
          class="item border-0 [object Object]"
          event="click"
          exactactiveclass="active"
          routertag="a"
          tag="div"
          target="_self"
        >
          <!---->
           
          <dl>
            <dt
              class=""
            >
              
              2. question
            
            </dt>
             
            <dd>
              <span>
                02/20/2018
              </span>
            </dd>
          </dl>
           
          <button-cmp-stub
            class="open-query"
            variant="outline-primary"
          >
            Open a query
          </button-cmp-stub>
        </b-list-group-item-stub>
      </b-list-group-stub>
    </b-card-stub>
  </b-card-group-stub>
</div>
`;
